Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -197,12 +197,21 @@ public override bool TryGetVariable(string name, out object value)
 
             context.Publish(completionRequestReceived);
 
-            var completionList =
-                GetCompletionList(
-                    requestCompletion.Code,
-                    requestCompletion.CursorPosition);
+            CommandCompletion completion = CommandCompletion.CompleteInput(requestCompletion.Code, requestCompletion.CursorPosition, null, _lazyPwsh.Value);
 
-            context.Publish(new CompletionRequestCompleted(completionList, requestCompletion));
+            var completionList = completion.CompletionMatches.Select(c => new CompletionItem(
+                displayText: c.CompletionText,
+                kind: c.ResultType.ToString(),
+                documentation: c.ToolTip
+            ));
+
+            context.Publish(new CompletionRequestCompleted(
+                completionList,
+                requestCompletion,
+                // PowerShell's replacement length is inclusive while Jupyter's ReplacementEndColumn is exclusive so we
+                // have to add one to the ReplaceLength for completions to be correct.
+                completion.ReplacementIndex,
+                completion.ReplacementLength + 1));
```

Comment:
`ReplacementLength` is the length of text to be replace starting from `ReplacementIndex`, right? But `cursor_end` is the cursor position where the replacement ends.
I think you need to calculate the `cursor_end` instead of using `length` directly.

Reply:
fixed!